### DISCUSSION TITLE
Kamal vs k0s+pulumi

### DISCUSSION THREAD

**Discussion by hrstoyanov** [basecamp/kamal#764] [General]
Hi all,
I am not rails/ruby developers, (not looking forward to become one). I am looking at Kamal  as an easy deployment platform only,  and trying to compare it to similarly simplified kubernetes platforms such as K0S/K3Sand Pulumi. On the surface, it looks like K0S+Pulumi give you the same level of simplification and productivity, yet being more scalable and standard.

Interesting in someone who has gone through the same decision making process, and decided to go with Kamal, and why. Even more interested in your opinion,  if you are not a rails developer (no bias)

**Comment by edzhelyov** [comment]
I think you will be comparing apples vs oranges here. Pulumi is a very generic tool for IaC, K0s and K3s are simplified Kubernetes. These are a bit low level tools that have *huge* learning curve.

In regard Kamal under the hood is Traefik load balancing network traffic to docker containers which is much simpler to understand. Then you have SSH access and running imperative commands on each host, much like Ansible, that are wrapping a blue-green deploy using Traefik. If this works for your case, then you are good to go.
